### PR TITLE
feat: refine promise usages in smart coffee machine scripts

### PR DESCRIPTION
**Pull request by fatadel**
Part of #331
Signed-off-by: fatadel <fatadel@gmail.com>

### CODE REVIEW

**Review by danielpeintner — CHANGES_REQUESTED** [review]
I am more in favor of nesting code with if/else so that I can be sure that a `resolve` or `reject` is the last possible command executed.

However if you want to keep your code style I suggest adding a `return` statement after `resolve`/`reject` if yet another `resolve`/`reject` can follow.

see https://stackoverflow.com/questions/32536049/do-i-need-to-return-after-early-resolve-reject

**Comment by fatadel** [comment]
> I am more in favor of nesting code with if/else so that I can be sure that a `resolve` or `reject` is the last possible command executed.
> 
> However if you want to keep your code style I suggest adding a `return` statement after `resolve`/`reject` if yet another `resolve`/`reject` can follow.
> 
> see https://stackoverflow.com/questions/32536049/do-i-need-to-return-after-early-resolve-reject

Done!

File: packages/core/test/ServerTest.ts [hunk 1]
```diff
@@ -612,6 +612,48 @@ class WoTServerTest {
         expect(thing).to.have.property("@context").to.deep.include({"@language": "xx"});
     }
 
+    @test async "should reject reading property without uriVariables when they are mandatory"() {
```
Comment (danielpeintner) [inline packages/core/test/ServerTest.ts:615]: Thank you @fatadel . You are right this is really an issue.
I will fix it in another PR to better keep track.

May I ask you to remove the test case in this pr. Sorry indeed and thanks!

Comment (fatadel) [reply packages/core/test/ServerTest.ts]: Sure, no worries!

**Review by danielpeintner — APPROVED** [review]
Looks good to me.

Whether you want to use `async` and `await` as mentioned by @relu91 [here](https://github.com/eclipse/thingweb.node-wot/issues/331#issuecomment-699088498) is up to you.

Just let me know whether you want to make these changes or you prefer keeping it as is.

**Comment by fatadel** [comment]
> Looks good to me.
> 
> Whether you want to use `async` and `await` as mentioned by @relu91 [here](https://github.com/eclipse/thingweb.node-wot/issues/331#issuecomment-699088498) is up to you.
> 
> Just let me know whether you want to make these changes or you prefer keeping it as is.

I would prefer to keep it as it is. `async/await` might make the code more understandable but for now it is consistent with other promise usages in this example (and also in others I think).

**Comment by danielpeintner** [comment]
Makes sense to me since we now have both possibilities in the examples directory (the counter uses async-await on request, see https://github.com/eclipse/thingweb.node-wot/pull/335)

**Pull request merged**